File: javascript/src/model/ConferencingInfoPhoneNumberInfoCountryInfo.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.RingCentralApi) {
      root.RingCentralApi = {};
    }
    root.RingCentralApi.ConferencingInfoPhoneNumberInfoCountryInfo = factory(root.RingCentralApi.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The ConferencingInfoPhoneNumberInfoCountryInfo model module.
   * @module model/ConferencingInfoPhoneNumberInfoCountryInfo
   * @version v1.0
   */

  /**
   * Constructs a new <code>ConferencingInfoPhoneNumberInfoCountryInfo</code>.
   * @alias module:model/ConferencingInfoPhoneNumberInfoCountryInfo
   * @class
   */
  var exports = function() {







  };

  /**
   * Constructs a <code>ConferencingInfoPhoneNumberInfoCountryInfo</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/ConferencingInfoPhoneNumberInfoCountryInfo} obj Optional instance to populate.
   * @return {module:model/ConferencingInfoPhoneNumberInfoCountryInfo} The populated <code>ConferencingInfoPhoneNumberInfoCountryInfo</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) { 
      obj = obj || new exports();

      if (data.hasOwnProperty('id')) {
        obj['id'] = ApiClient.convertToType(data['id'], 'String');
      }
      if (data.hasOwnProperty('uri')) {
        obj['uri'] = ApiClient.convertToType(data['uri'], 'String');
      }
      if (data.hasOwnProperty('callingCode')) {
        obj['callingCode'] = ApiClient.convertToType(data['callingCode'], 'String');
      }
      if (data.hasOwnProperty('emergencyCalling')) {
        obj['emergencyCalling'] = ApiClient.convertToType(data['emergencyCalling'], 'Boolean');
      }
      if (data.hasOwnProperty('isoCode')) {
        obj['isoCode'] = ApiClient.convertToType(data['isoCode'], 'String');
      }
      if (data.hasOwnProperty('name')) {
        obj['name'] = ApiClient.convertToType(data['name'], 'String');
      }
    }
    return obj;
  }


  /**
   * Internal identifier of a country
   * @member {String} id
   */
  exports.prototype['id'] = undefined;

  /**
   * Canonical URI of a country
   * @member {String} uri
   */
  exports.prototype['uri'] = undefined;

  /**
   * Country calling code defined by ITU-T recommendations E.123 and E.164, see Calling Codes
   * @member {String} callingCode
   */
  exports.prototype['callingCode'] = undefined;

  /**
   * Emergency calling feature availability/emergency address requirement indicator
   * @member {Boolean} emergencyCalling
   */
  exports.prototype['emergencyCalling'] = undefined;

  /**
   * Country code according to the ISO standard, see ISO 3166
   * @member {String} isoCode
   */
  exports.prototype['isoCode'] = undefined;

  /**
   * Official name of a country
   * @member {String} name
   */
  exports.prototype['name'] = undefined;




  return exports;
}));

File: csharp/src/main/csharp/IO/Swagger/Model/ConferencingInfoPhoneNumberInfoCountryInfo.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace IO.Swagger.Model
{
    /// <summary>
    /// 
    /// </summary>
    [DataContract]
    public partial class ConferencingInfoPhoneNumberInfoCountryInfo :  IEquatable<ConferencingInfoPhoneNumberInfoCountryInfo>
    { 
    
        /// <summary>
        /// Initializes a new instance of the <see cref="ConferencingInfoPhoneNumberInfoCountryInfo" /> class.
        /// Initializes a new instance of the <see cref="ConferencingInfoPhoneNumberInfoCountryInfo" />class.
        /// </summary>
        /// <param name="Id">Internal identifier of a country.</param>
        /// <param name="Uri">Canonical URI of a country.</param>
        /// <param name="CallingCode">Country calling code defined by ITU-T recommendations E.123 and E.164, see Calling Codes.</param>
        /// <param name="EmergencyCalling">Emergency calling feature availability/emergency address requirement indicator.</param>
        /// <param name="IsoCode">Country code according to the ISO standard, see ISO 3166.</param>
        /// <param name="Name">Official name of a country.</param>

        public ConferencingInfoPhoneNumberInfoCountryInfo(string Id = null, string Uri = null, string CallingCode = null, bool? EmergencyCalling = null, string IsoCode = null, string Name = null)
        {
            this.Id = Id;
            this.Uri = Uri;
            this.CallingCode = CallingCode;
            this.EmergencyCalling = EmergencyCalling;
            this.IsoCode = IsoCode;
            this.Name = Name;
            
        }
        
    
        /// <summary>
        /// Internal identifier of a country
        /// </summary>
        /// <value>Internal identifier of a country</value>
        [DataMember(Name="id", EmitDefaultValue=false)]
        public string Id { get; set; }
    
        /// <summary>
        /// Canonical URI of a country
        /// </summary>
        /// <value>Canonical URI of a country</value>
        [DataMember(Name="uri", EmitDefaultValue=false)]
        public string Uri { get; set; }
    
        /// <summary>
        /// Country calling code defined by ITU-T recommendations E.123 and E.164, see Calling Codes
        /// </summary>
        /// <value>Country calling code defined by ITU-T recommendations E.123 and E.164, see Calling Codes</value>
        [DataMember(Name="callingCode", EmitDefaultValue=false)]
        public string CallingCode { get; set; }
    
        /// <summary>
        /// Emergency calling feature availability/emergency address requirement indicator
        /// </summary>
        /// <value>Emergency calling feature availability/emergency address requirement indicator</value>
        [DataMember(Name="emergencyCalling", EmitDefaultValue=false)]
        public bool? EmergencyCalling { get; set; }
    
        /// <summary>
        /// Country code according to the ISO standard, see ISO 3166
        /// </summary>
        /// <value>Country code according to the ISO standard, see ISO 3166</value>
        [DataMember(Name="isoCode", EmitDefaultValue=false)]
        public string IsoCode { get; set; }
    
        /// <summary>
        /// Official name of a country
        /// </summary>
        /// <value>Official name of a country</value>
        [DataMember(Name="name", EmitDefaultValue=false)]
        public string Name { get; set; }
    
        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class ConferencingInfoPhoneNumberInfoCountryInfo {\n");
            sb.Append("  Id: ").Append(Id).Append("\n");
            sb.Append("  Uri: ").Append(Uri).Append("\n");
            sb.Append("  CallingCode: ").Append(CallingCode).Append("\n");
            sb.Append("  EmergencyCalling: ").Append(EmergencyCalling).Append("\n");
            sb.Append("  IsoCode: ").Append(IsoCode).Append("\n");
            sb.Append("  Name: ").Append(Name).Append("\n");
            
            sb.Append("}\n");
            return sb.ToString();
        }
  
        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="obj">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object obj)
        {
            // credit: http://stackoverflow.com/a/10454552/677735
            return this.Equals(obj as ConferencingInfoPhoneNumberInfoCountryInfo);
        }

        /// <summary>
        /// Returns true if ConferencingInfoPhoneNumberInfoCountryInfo instances are equal
        /// </summary>
        /// <param name="other">Instance of ConferencingInfoPhoneNumberInfoCountryInfo to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(ConferencingInfoPhoneNumberInfoCountryInfo other)
        {
            // credit: http://stackoverflow.com/a/10454552/677735
            if (other == null)
                return false;

            return 
                (
                    this.Id == other.Id ||
                    this.Id != null &&
                    this.Id.Equals(other.Id)
                ) && 
                (
                    this.Uri == other.Uri ||
                    this.Uri != null &&
                    this.Uri.Equals(other.Uri)
                ) && 
                (
                    this.CallingCode == other.CallingCode ||
                    this.CallingCode != null &&
                    this.CallingCode.Equals(other.CallingCode)
                ) && 
                (
                    this.EmergencyCalling == other.EmergencyCalling ||
                    this.EmergencyCalling != null &&
                    this.EmergencyCalling.Equals(other.EmergencyCalling)
                ) && 
                (
                    this.IsoCode == other.IsoCode ||
                    this.IsoCode != null &&
                    this.IsoCode.Equals(other.IsoCode)
                ) && 
                (
                    this.Name == other.Name ||
                    this.Name != null &&
                    this.Name.Equals(other.Name)
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            // credit: http://stackoverflow.com/a/263416/677735
            unchecked // Overflow is fine, just wrap
            {
                int hash = 41;
                // Suitable nullity checks etc, of course :)
                
                if (this.Id != null)
                    hash = hash * 59 + this.Id.GetHashCode();
                
                if (this.Uri != null)
                    hash = hash * 59 + this.Uri.GetHashCode();
                
                if (this.CallingCode != null)
                    hash = hash * 59 + this.CallingCode.GetHashCode();
                
                if (this.EmergencyCalling != null)
                    hash = hash * 59 + this.EmergencyCalling.GetHashCode();
                
                if (this.IsoCode != null)
                    hash = hash * 59 + this.IsoCode.GetHashCode();
                
                if (this.Name != null)
                    hash = hash * 59 + this.Name.GetHashCode();
                
                return hash;
            }
        }

    }
}
